compute/cloud-client/src/main/java/compute/reservation/CreateInstanceNotConsumeReservation.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.reservation;
+
+// [START compute_instance_not_consume_reservation]
+
+import static com.google.cloud.compute.v1.ReservationAffinity.ConsumeReservationType.NO_RESERVATION;
+
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.compute.v1.AttachedDisk;
+import com.google.cloud.compute.v1.AttachedDiskInitializeParams;
+import com.google.cloud.compute.v1.InsertInstanceRequest;
+import com.google.cloud.compute.v1.Instance;
+import com.google.cloud.compute.v1.InstancesClient;
+import com.google.cloud.compute.v1.NetworkInterface;
+import com.google.cloud.compute.v1.Operation;
+import com.google.cloud.compute.v1.ReservationAffinity;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateInstanceNotConsumeReservation {
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+    String zone = "us-central1-a";
+    // Name of the VM instance you want to query.
+    String instanceName = "YOUR_INSTANCE_NAME";
+
+    createInstanceNotConsumeReservation(projectId, zone, instanceName);
+  }
+
+  // Create a virtual machine that explicitly doesn't consume reservations
+  public static void createInstanceNotConsumeReservation(
+      String project, String zone, String instanceName)
+      throws IOException, InterruptedException, ExecutionException, TimeoutException {
+    // Below are sample values that can be replaced.
+    // machineType: machine type of the VM being created.
+    // *   This value uses the format zones/{zone}/machineTypes/{type_name}.
+    // *   For a list of machine types, see https://cloud.google.com/compute/docs/machine-types
+    // sourceImage: path to the operating system image to mount.
+    // *   For details about images you can mount, see https://cloud.google.com/compute/docs/images
+    // diskSizeGb: storage size of the boot disk to attach to the instance.
+    // networkName: network interface to associate with the instance.
```

Comment:
if these are values that you want a user to provide, refactor your code to pass these values as arguments to the method call. Otherwise, remove this large comment because it is distracting and it repeats after the documentation that is expected to use this code snippet.

Reply:
The amount of values that have to be provided while creating new VM instance is absurd. I have *some* of those present in Python samples and it's really difficult to read when you have a method with >10 parameters.

IMO this approach is much more manageable and makes the sample easier to use by customers, when they have sensible values already provided.